src/transaction.js

```diff
@@ -120,8 +120,13 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
   }
 
   if (hasWitnesses) {
+    var isNull = true
     for (i = 0; i < vinLen; ++i) {
       tx.ins[i].witness = readVector()
+      isNull = isNull && tx.ins[i].witness.length === 0
+    }
+    if (isNull) {
+      throw new Error('Transaction has superfluous witness data')
```

Comment:
Nevermind... from https://bitcoincore.org/en/segwit_wallet_dev/

> 
If all txins in a transaction are not associated with any witness data, the transaction MUST be serialized in the original transaction format, without marker, flag, and witness. For example, if none of the txins are coming from segwit UTXO, it MUST be serialized in the original transaction format. (exception: coinbase transaction)